libkatana_python_native/src/Reductions.cpp

```diff
@@ -0,0 +1,90 @@
+#include <katana/Reduction.h>
+#include <katana/python/Conventions.h>
+#include <katana/python/NumbaSupport.h>
+#include <katana/python/PythonModuleInitializers.h>
+#include <katana/python/TemplateSupport.h>
+#include <pybind11/pybind11.h>
+
+namespace py = pybind11;
+
+namespace {
+
+// TODO(amp): Use template template parameters once we can rely on having them.
+struct ForGAccumulator {
+  template <typename T>
+  using type = katana::GAccumulator<T>;
+};
+
+struct ForReduceMin {
+  template <typename T>
+  using type = katana::GReduceMin<T>;
+};
+
+struct ForReduceMax {
+  template <typename T>
+  using type = katana::GReduceMax<T>;
+};
+
+struct ForReduceLogicalOr {
+  template <typename T>
+  using type = katana::GReduceLogicalOr;
+};
```

Comment:
I understand the need for templating for the others, but no one is using `typename T` here, so why is it needed?

Reply:
It's needed to match the type level interface provided by the structs above since they are all used in the same code below.